### ISSUE TITLE
EBS Volume Filter Not Working- Incorrect Output results

### ISSUE THREAD

**Issue by aakifshaikh** [cloud-custodian/cloud-custodian#6070]
**Describe the bug**
We have a written cloud custodian policy to look for non-root unencrypted EBS volume. Below is the actual written policy. However, it looks like the policy is even considering the root volume in its filter. When we query our SIEM solution, we get to see that this policy filtering is incorrect and not giving the right output results. We verified our results even going into AWS console and manually verifying it.

**To Reproduce**
See the written policy below

**Expected behavior**
We expected the custodian to produce the result to include ONLY non-root EBS volumes which are not encrypted.
We have a filter written in the policy to ignore the root volumes.

```yaml
vars:
  sec-tags: &sec-tags
    owner: XYZ
    service: cloudcustodian

policies:

- name: sec-n-ebs-volume-notify-if-unencrypted
  comment: |
    Find all non-root, unencrypted, EBS volumes.
    With exemption tagging in place, cloud custodian will not take any action, ignore that resource and jump onto the next one.
    Exemption tag can be used by submitting the OPENSEC risk ticket.
  resource: aws.ebs
  filters:
    - "tag:c7n_nonroot_ebs_volume_encryption_exempt": absent
    - Encrypted: false
    - not:
      - type: value
        key: "Attachments[].Device"
        op: regex
        value: ^/dev/(sda1|xvda)$
  mode:
    schedule: "rate(3 days)"
    type: periodic
    execution-options:
      output_dir: s3://X/X/X/X/X/X/
    tags: *sec-tags
```

 - custodian version --0.9.2.0

**Additional context**
- Verified the custodian output manually within AWS console and seems that custodian is giving incorrect results.
- Other developers have also verified this at their end.
- This policy was working fine before- we recently noticed this incorrectness.

**Comment by PratMis** [comment]
Does the root volume that's getting filtered not have `Attachments[].Device: ^/dev/(sda1|xvda)$` regex?

**Comment by kapilt** [comment]
are you taking into account the default cache period?

**Comment by kapilt** [comment]
also you can do the same from the ec2 side with an ebs volume which has builtin foptions or filtering out the root, this seems like an issue with your regex filter.

**Comment by Bunnikins** [comment]
Hi, sorry to be a pain but this seems to be very similar to my reported issue (https://github.com/cloud-custodian/cloud-custodian/issues/6079) - I'm only getting responses on root volume as well. Likely to be raising another issue in parallel as ec2 also appears to only check the subnet on primary enis.
